### PR TITLE
Added global design tokens - spacing, duration, curves

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
# Changes
- Added spacing type
- Added horizontal and vertical spacing to theme
- Added duration type to theme
- Added curve type to theme
- Defined constants for new theme types based on figma values
- Initialized all themes with constants

# Issues
Updates #22231

### CODE REVIEW

File: packages/react-theme/etc/react-theme.api.md [hunk 1]
```diff
@@ -352,6 +376,21 @@ export type FontWeightTokens = {
     fontWeightSemibold: number;
 };
 
+// @public (undocumented)
+export type HorizontalSpacingTokens = {
+    spacingHorizontalNone: string;
+    spacingHorizontalXXS: string;
```
Comment (GeoffCoxMSFT) [inline packages/react-theme/etc/react-theme.api.md:382]: @miroslavstastny I followed the figma naming for the spacing from the figma.  When I needed to add the spacingHorizontal and spacingVertical prefixes, they are a little hard to read.  Do you think we should have an underscore between the prefix and the size (e.g. spacingHorizontal_XXS) ?  Or maybe you have a better way to name them?

File: packages/react-theme/etc/react-theme.api.md [hunk 2]
```diff
@@ -389,6 +428,21 @@ export type ShadowTokens = {
     shadow64: string;
 };
 
+// @public (undocumented)
+export type SpacingTokens = {
```
Comment (GeoffCoxMSFT) [inline packages/react-theme/etc/react-theme.api.md:432]: @miroslavstastny The spacing per the figma has the horizontal and vertical sets of these on the alias sheet.  I assume this means that they are the ones exposed as tokens on the theme and the spacing tokens are just the default seed for both of them.  You can see I use the spacing tokens constant to initialize the horizontal/vertical in the theme.

Do you think this type should be exported?

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                              |   Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                      Change |
| :------------------------------------------------------------------------------------------------------------- | -------------------------: | --------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsAllThemes.fixture.js'>Teams: all themes</abbr>   | `29.676 kB`<br />`6.61 kB` | `30.905 kB`<br />`6.993 kB` | `1.229 kB` <br />`383 B`  |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsLightTheme.fixture.js'>Teams: Light theme</abbr> |  `18.492 kB`<br />`5.3 kB` | `19.632 kB`<br />`5.685 kB` |  `1.14 kB` <br />`385 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                   |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> | `73.754 kB`<br />`22.494 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                          | `45.196 kB`<br />`13.126 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                             |  `20.965 kB`<br />`6.605 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                               |  `21.918 kB`<br />`6.919 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                             |  `22.005 kB`<br />`6.582 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                          |  `32.408 kB`<br />`8.986 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                          | `38.635 kB`<br />`10.146 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                  |  `34.175 kB`<br />`9.567 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                | `41.431 kB`<br />`10.865 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                              |   `43.31 kB`<br />`9.921 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                      | `57.884 kB`<br />`16.524 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                | `53.169 kB`<br />`15.238 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                    |   `7.756 kB`<br />`3.299 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                    |   `9.321 kB`<br />`3.816 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                  |   `7.728 kB`<br />`3.323 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox</abbr>                                    | `60.564 kB`<br />`20.751 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                       |   `15.455 kB`<br />`5.57 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                             |  `10.179 kB`<br />`3.995 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                             |  `21.775 kB`<br />`7.228 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                             |   `8.441 kB`<br />`3.541 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                               |  `11.176 kB`<br />`4.545 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                       | `100.693 kB`<br />`30.67 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                          |    `6.272 kB`<br />`2.17 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePopper</abbr>                               |   `23.21 kB`<br />`8.084 kB` |
| <samp>react-priority-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                            |  `10.792 kB`<br />`4.125 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                        |  `14.079 kB`<br />`5.274 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                          |  `16.616 kB`<br />`6.281 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                          |  `25.619 kB`<br />`8.287 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                              | `42.003 kB`<br />`11.886 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                       |  `17.224 kB`<br />`5.856 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                          |  `24.333 kB`<br />`8.019 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                              |  `10.867 kB`<br />`4.269 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                            |   `14.183 kB`<br />`4.61 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                    |   `20.672 kB`<br />`7.07 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/SingleThemeToken.fixture.js'>Single theme token import</abbr>              |           `69 B`<br />`89 B` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                       | `42.907 kB`<br />`14.761 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1cb4002d46236c724a274ff5abafc1c1e99397e2'>1cb4002d46236c724a274ff5abafc1c1e99397e2</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-theme/src/global/spacings.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import type { HorizontalSpacingTokens, SpacingTokens, VerticalSpacingTokens } from '../types';
+
+export const spacings: SpacingTokens = {
```
Comment (micahgodbolt) [inline packages/react-theme/src/global/spacings.ts:3]: do these raw values need to be exported? is there a precedent for making them available?

Comment (micahgodbolt) [reply packages/react-theme/src/global/spacings.ts:3]: oh, i guess i'm assuming they are exported top level, which might not be the case

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/Avatar-mount_base.html">1293</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/Avatar-mount.html">1305</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/Button-mount_base.html">826</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/Button-mount.html">798</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/FluentProvider-mount_base.html">2246</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/FluentProvider-mount.html">2344</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/FluentProviderWithTheme-mount_base.html">410</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/FluentProviderWithTheme-mount.html">429</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">351</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">377</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">458</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">490</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/MakeStyles-mount_base.html">2001</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22607/perf-test-react-components/MakeStyles-mount.html">2046</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by micahgodbolt — APPROVED** [review]

**Review by miroslavstastny — APPROVED** [review]
Thanks for adding the tokens!
We also need to add stories, I've created a separate tracking issue #22690.

File: packages/react-theme/etc/react-theme.api.md [hunk 4]
```diff
@@ -356,6 +380,21 @@ export type FontWeightTokens = {
...
+    spacingHorizontalXXS: string;
+    spacingHorizontalXS: string;
+    spacingHorizontalSNudge: string;
+    spacingHorizontalS: string;
+    spacingHorizontalMNudge: string;
+    spacingHorizontalM: string;
+    spacingHorizontalL: string;
+    spacingHorizontalXL: string;
+    spacingHorizontalXXL: string;
+    spacingHorizontalXXXl: string;
```
Comment (GeoffCoxMSFT) [inline packages/react-theme/etc/react-theme.api.md:395]: This should end in XXXL.

**Pull request merged**